src/tele_out/task_fetcher.rs

```diff
@@ -34,64 +32,52 @@ impl TaskFetcher {
         }
     }
 
-    async fn reset_submitting(&self) -> Result<(), anyhow::Error> {
-        let stmt = format!("update {} set status = $1 where status = $2", models::tablenames::L2_BLOCK);
-        sqlx::query(&stmt)
-            .bind(models::l2_block::BlockStatus::Submitting)
-            .bind(models::l2_block::BlockStatus::Uncommited)
-            .execute(&self.connpool)
-            .await?;
-        Ok(())
-    }
-
-    async fn run_inner(&self, tx: &Sender<ContractCall>) -> Result<(), anyhow::Error> {
+    async fn run_inner(&mut self, tx: &Sender<ContractCall>) -> Result<(), anyhow::Error> {
         let mut db_tx = self.connpool.begin().await?;
 
+        #[derive(sqlx::FromRow, Debug, Clone)]
+        struct Task {
+            block_id: i64,
+            public_input: Vec<u8>,
+            proof: Vec<u8>,
+        }
+
         let query: &'static str = const_format::formatcp!(
             r#"
-            select
-                t.task_id as task_id, 
-                t.circuit as circuit, 
-                t.block_id as block_id, 
-                t.input as input, 
-                t.output as output, 
-                t.witness as witness, 
-                t.public_input as public_input, 
-                t.proof as proof, 
-                t.status as status, 
-                t.prover_id as prover_id, 
-                t.created_time as created_time, 
-                t.updated_time as updated_time
+            select t.block_id     as block_id,
+                   t.public_input as public_input,
+                   t.proof        as proof
             from {} t
-                inner join {} b on t.block_id = b.block_id
-            where b.status = $1 and t.status = $2
-            ORDER BY t.block_id ASC
-            LIMIT 1"#,
+                     inner join {} l2b
+                                on t.block_id = l2b.block_id
+            where t.block_id < coalesce((select block_id
+                                         from task
+                                         where status <> 'proved'
+                                         order by block_id
+                                         limit 1), 0)
+              and t.block_id > $1
+              and t.status = 'proved' -- defense filter
+              and l2b.status = 'uncommited'
+            order by t.block_id
```

Comment:
better to make ` order by t.block_id ASC` explicitly?

Reply:
asc is always the default order, this is redundant